src/plugins/btrfs.c

```diff
@@ -454,47 +455,12 @@ gboolean bd_btrfs_delete_subvolume (const gchar *mountpoint, const gchar *name,
  * Tech category: %BD_BTRFS_TECH_SUBVOL-%BD_BTRFS_TECH_MODE_QUERY
  */
 guint64 bd_btrfs_get_default_subvolume_id (const gchar *mountpoint, GError **error) {
-    GRegex *regex = NULL;
-    GMatchInfo *match_info = NULL;
-    gboolean success = FALSE;
-    gchar *output = NULL;
-    gchar *match = NULL;
+    enum btrfs_util_error err;
     guint64 ret = 0;
-    const gchar *argv[5] = {"btrfs", "subvol", "get-default", mountpoint, NULL};
-
-    if (!check_deps (&avail_deps, DEPS_BTRFS_MASK, deps, DEPS_LAST, &deps_check_lock, error) ||
-        !check_module_deps (&avail_module_deps, MODULE_DEPS_BTRFS_MASK, module_deps, MODULE_DEPS_LAST, &deps_check_lock, error))
-        return 0;
-
-    regex = g_regex_new ("ID (\\d+) .*", 0, 0, error);
-    if (!regex) {
-        bd_utils_log_format (BD_UTILS_LOG_WARNING, "Failed to create new GRegex");
-        /* error is already populated */
-        return 0;
-    }
-
-    success = bd_utils_exec_and_capture_output (argv, NULL, &output, error);
-    if (!success) {
-        g_regex_unref (regex);
-        return 0;
-    }
-
-    success = g_regex_match (regex, output, 0, &match_info);
-    if (!success) {
-        g_set_error (error, BD_BTRFS_ERROR, BD_BTRFS_ERROR_PARSE, "Failed to parse subvolume's ID");
-        g_regex_unref (regex);
-        g_match_info_free (match_info);
-        g_free (output);
-        return 0;
-    }
-
-    match = g_match_info_fetch (match_info, 1);
-    ret = g_ascii_strtoull (match, NULL, 0);
 
-    g_free (match);
-    g_match_info_free (match_info);
-    g_regex_unref (regex);
-    g_free (output);
+    err = btrfs_util_get_default_subvolume(mountpoint, &ret);
+    if (err)
+        g_set_error (error, BD_BTRFS_ERROR, BD_BTRFS_ERROR_PARSE, btrfs_util_strerror(err));
```

Comment:
I think we can afford changing error codes a little bit, if needed. While it would be nice to keep most of them consistent, I imagine the code path would be so different anyway and it would be good to distinguish between minor errors (i.e. "device open error", "parse error", "invalid argument", etc.)

Reply:
Agreed, currently the error is rather strange when given a path which is not mounted. So this needs more work.
```
AssertionError: ".*(can't|cannot) access.*" does not match "g-bd-btrfs-error-quark: Could not open (2)" 
```

Reply:
It's a C style API so the test of the error was in errno.

Reply:
Different error messages are fine, there's generally no guarantee of stability and clients should never match any strings (it's okay for tests where it is intentional) - that's what the error codes are for.